src/operators.jl

```diff
@@ -83,120 +127,152 @@ function Base.:^(lhs::Union{AbstractVariableRef,GenericAffExpr}, rhs::Integer)
 end
 Base.:^(lhs::Union{AbstractVariableRef,GenericAffExpr}, rhs::Number) = error("Only exponents of 0, 1, or 2 are currently supported. Are you trying to build a nonlinear problem? Make sure you use @NLconstraint/@NLobjective.")
 # GenericAffExpr--AbstractVariableRef
-Base.:+(lhs::GenericAffExpr{C,V}, rhs::V) where {C,V<:JuMPTypes} = GenericAffExpr{C,V}(vcat(lhs.vars,rhs),vcat(lhs.coeffs,one(C)), lhs.constant)
-Base.:-(lhs::GenericAffExpr{C,V}, rhs::V) where {C,V<:JuMPTypes} = GenericAffExpr{C,V}(vcat(lhs.vars,rhs),vcat(lhs.coeffs,-one(C)),lhs.constant)
-# Don't fall back on AbstractVariableRef*GenericAffExpr to preserve lhs/rhs consistency (appears in printing)
-function Base.:*(lhs::GenericAffExpr{T,V}, rhs::V) where {T,V}
-    n = length(lhs.vars)
+function Base.:+(lhs::GenericAffExpr{C,V}, rhs::V) where {C, V <: AbstractVariableRef}
```

Comment:
I don't think we need `V <: AbstractVariableRef` since this is already implied by the fact that `V` is the second parameter of `GenericAffExpr{C,V}`

Reply:
We specify `V <: AbstractVariableRef` in lots of places in the overloads for `AbstractVariableRef` above, why shouldn't it be here?

Reply:
As a second thought, it's probably better to leave it. We might change the method at some point and it might be needed then so it's safer to be explicit